lib/shopify/resources/metafield.ex

```diff
@@ -15,4 +32,53 @@ defmodule Shopify.Metafield do
     :value_type,
     :updated_at
   ]
+
+  @doc false
+  def empty_resource do
+    %Metafield{
+      key: "",
+      value: "",
+      value_type: "string",
+      namespace: ""
+    }
+  end
+
+  [:blogs, :collections, :customers, :draft_orders, :orders, :pages, :products]
+  |> Enum.each(fn (name) ->
+    def unquote(name)(session, parent_id, params \\ %{}) do
+      session
+      |> Request.new(metafield_url(Atom.to_string(unquote(name)), parent_id), params, plural_resource())
+      |> Client.get()
+    end
+  end)
+
+  [[:blogs, :articles], [:products, :variants]]
+  |> Enum.each(fn (names) ->
+    parent_name = Enum.at(names, 0)
+    child_name  = Enum.at(names, 1)
+    def unquote(child_name)(session, parent_id, child_id, params \\ %{}) do
+      session
+      |> Request.new(metafield_url(Atom.to_string(unquote(parent_name)), parent_id,Atom.to_string(unquote(child_name)), child_id), params, plural_resource())
```

Comment:
Same here, using strings would remove the castings and make the line a little shorter 😃